apps/wallet/src/ui/app/pages/swap/SwapPageForm.tsx

```diff
@@ -0,0 +1,460 @@
+				)}
+			>
+				{rawValue && !isRefetching ? (
+					<>
+						<Text variant="body" weight="semibold" color="steel-darker">
+							{rawValue}
+						</Text>
+						<Text variant="body" weight="semibold" color="steel">
+							{quotedAssetSymbol}
+						</Text>
+					</>
+				) : (
+					<Text variant="body" weight="semibold" color="steel">
+						--
+					</Text>
+				)}
+			</div>
+			{rawValue && (
+				<div className="ml-3">
+					<DescriptionItem
+						title={
+							<Text variant="bodySmall" color="steel-dark">
+								{isRefetching ? '--' : getUSDCurrency(rawValueQuoteToUsd)}
+							</Text>
+						}
+					>
+						<div className="flex gap-1 items-center">
+							<Text variant="bodySmall" weight="medium" color="steel-dark">
+								1 {activeCoinData?.symbol} = {isRefetching ? '--' : averagePriceAsString}{' '}
+								{quotedAssetSymbol}
+							</Text>
+							<IconButton
+								icon={<Refresh16 className="h-4 w-4 text-steel-dark hover:text-hero-dark" />}
+								onClick={() => refetch()}
+								loading={isRefetching}
+							/>
+						</div>
+					</DescriptionItem>
+
+					<div className="h-px w-full bg-hero-darkest/10 my-3" />
+
+					<MaxSlippage />
+				</div>
+			)}
+		</div>
+	);
+}
+
+function GasFeeSection() {
+	const { values, isValid } = useFormikContext<FormValues>();
+	const [searchParams] = useSearchParams();
+
+	const activeCoinType = searchParams.get('type');
+
+	const { data: activeCoinData } = useCoinMetadata(activeCoinType);
+
+	const amount = values.amount;
+
+	const estimatedFess = useMemo(() => {
+		if (!amount || !isValid) {
+			return null;
+		}
+
+		return new BigNumber(amount).times(FEES_PERCENTAGE);
+	}, [amount, isValid]);
+
+	const estimatedFessAsBigInt = estimatedFess ? new BigNumber(estimatedFess) : null;
+
+	// TODO: need to handle for all coins
+	const { rawValue } = useSuiBalanceInUSDC(estimatedFessAsBigInt);
+
+	const formattedEstimatedFees = getUSDCurrency(rawValue);
+
+	return (
+		<div className="flex flex-col border border-hero-darkest/20 rounded-xl p-5 gap-4 border-solid">
+			<DescriptionItem
+				title={
+					<Text variant="bodySmall" weight="medium" color="steel-dark">
+						Fees ({FEES_PERCENTAGE}%)
+					</Text>
+				}
+			>
+				<Text variant="bodySmall" weight="medium" color="steel-darker">
+					{estimatedFess
+						? `${estimatedFess.toLocaleString()} ${activeCoinData?.symbol} (${formattedEstimatedFees})`
+						: '--'}
+				</Text>
+			</DescriptionItem>
+
+			<div className="bg-gray-40 h-px w-full" />
+
+			<DescriptionItem
+				title={
+					<Text variant="bodySmall" weight="medium" color="steel-dark">
+						Estimated Gas Fee
+					</Text>
+				}
+			>
+				<Text variant="bodySmall" weight="medium" color="steel-darker">
+					--
+				</Text>
+			</DescriptionItem>
+		</div>
+	);
+}
+
+function getSwapPageAtcText(fromSymbol: string, quoteAssetType: string) {
+	const toSymbol =
+		Object.entries(coinsMap).find(([key, value]) => value === quoteAssetType)?.[0] || '';
```

Comment:
Why this is not coin metadata and we are doing this coinMap search? It seems that we could use `useCoinMetadata` to get the symbol for the `quoteAssetType`

Reply:
This is to use for ATC text, this does not need to be safe. We are passing the symbols from useCoinMetaData